config/karma.saucelabs.js

```diff
@@ -156,7 +159,8 @@ module.exports = function(config) {
       accessKey: process.env.SAUCE_ACCESS_KEY,
       startConnect: true,
       connectOptions: {
-        noSslBumpDomains: 'all'
+        // https://wiki.saucelabs.com/display/DOCS/Troubleshooting+Sauce+Connect
```

Comment:
Perhaps:

Realtime Database uses WebSockets to connect to firebaseio.com so we have to set noSslBumpDomains. See https://wiki.saucelabs.com/display/DOCS/Troubleshooting+Sauce+Connect for more details.

Reply:
Ooops, I didn't mean to publish this...

Theoretically, we only need to whitelist *.firebaseio.com instead of all, but I've noticed that requests to firestore.googleapis.com increased which maybe the cause of many failures in Safari and Edge.

I'll try a few more things and finalize my code change soon.

Reply:
Didn't figure out the root cause behind it. Only noticed that test time can easily exceed 30 mins with only SSL Bumping for '*.firebaseio.com' disabled, rather than ~22mins in case of 'all'.